jiant/pytorch_transformers_interface/modules.py

```diff
@@ -0,0 +1,300 @@
+        # The AllenNLP indexer adds a '@@UNKNOWN@@' token to the
+        # beginning of the vocabulary, *and* treats that as index 1 (index 0 is
+        # reserved for padding).
+        ids[ids == 0] = self._pad_id + 2  # Shift the indices that were at 0 to become 2.
+        # Index 1 should never be used since the BERT WPM uses its own
+        # unk token, and handles this at the string level before indexing.
+        assert (ids > 1).all()
+        ids -= 2  # shift indices to match BERT wordpiece embeddings
+
+        if self.embeddings_mode not in ["none", "top"]:
+            # This is redundant with the lookup inside BertModel,
+            # but doing so this way avoids the need to modify the BertModel
+            # code.
+            # Extract lexical embeddings
+            lex_seq = self.model.embeddings.word_embeddings(ids)
+            lex_seq = self.model.embeddings.LayerNorm(lex_seq)
+            output_seq = None  # dummy; should not be accessed.
+            hidden_states = None  # dummy; should not be accessed.
+            # following our use of the OpenAI model, don't use dropout for
+            # probing. If you would like to use dropout, consider applying
+            # later on in the SentenceEncoder (see models.py).
+            #  h_lex = self.model.embeddings.dropout(embeddings)
+        else:
+            lex_seq = None  # dummy; should not be accessed.
+
+        if self.embeddings_mode != "only":
+            # encoded_layers is a list of layer activations, each of which is
+            # <float32> [batch_size, seq_len, output_dim]
+            token_types = self.get_seg_ids(ids)
+            output_seq, output_pooled_vec, hidden_states = self.model(
+                ids, token_type_ids=token_types, attention_mask=mask
+            )
+
+        # <float32> [batch_size, var_seq_len, output_dim]
+        return self.prepare_output(output_seq, lex_seq, hidden_states)
+
+
+class XLNetEmbedderModule(PytorchTransformersEmbedderModule):
+    """ Wrapper for XLNet module to fit into jiant APIs. """
+
+    def __init__(self, args):
+
+        super(XLNetEmbedderModule, self).__init__(args)
+
+        self.model = pytorch_transformers.XLNetModel.from_pretrained(
+            args.input_module, cache_dir=self.cache_dir, output_hidden_states=True
+        )
+
+        tokenizer = pytorch_transformers.XLNetTokenizer.from_pretrained(
+            args.input_module, cache_dir=self.cache_dir, do_lower_case="uncased" in args.tokenizer
+        )  # TODO: Speed things up slightly by reusing the previously-loaded tokenizer.
+        self._sep_id = tokenizer.convert_tokens_to_ids("<sep>")
+        self._cls_id = tokenizer.convert_tokens_to_ids("<cls>")
+        self._pad_id = tokenizer.convert_tokens_to_ids("<pad>")
+        self._unk_id = tokenizer.convert_tokens_to_ids("<unk>")
+
+        self.parameter_setup(args)
+
+        self._SEG_ID_CLS = 2
+        self._SEG_ID_SEP = 3
+
+    def apply_boundary_tokens(s1, s2=None):
```

Comment:
This might crash without `self` as the first argument, but probably better to just make this static instead?

e.g.
```
@staticmethod
def apply_boundary_tokens(s1, s2=None):
   ...
```

(and similarly with the BERT version)